internal/app/admin/http/cache.go

```diff
@@ -151,31 +147,26 @@ func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		cacheKey := getParam(req.URL.Path)
 		c := orchestrator.Orchestrator.GetReadOnlyCache(*o)
-		var keysToPrint []string
-
-		// If wildcard suffix provided, output all cache entries that match given prefix.
-		// If no key is provided, output the entire cache.
-		if hasWildcardSuffix(cacheKey) {
-			// Retrieve all keys
-			allKeys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
-				return
-			}
+		keysToPrint, err := getRelevantKeys(o, cacheKey, w)
+		if err.Message == "" {
+			printCacheEntries(keysToPrint, c, w)
+		}
+	}
+}
 
-			// Find keys that match prefix of wildcard
-			rootCacheKeyName := strings.TrimSuffix(cacheKey, "*")
-			for potentialMatchKey := range allKeys {
-				if strings.HasPrefix(potentialMatchKey, rootCacheKeyName) {
-					keysToPrint = append(keysToPrint, potentialMatchKey)
-				}
+func clearCacheHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		if req.Method == "POST" {
+			cacheKey := getParam(req.URL.Path)
+			c := orchestrator.Orchestrator.GetCache(*o)
+			keysToClear, err := getRelevantKeys(o, cacheKey, w)
+			if err.Message == "" {
+				clearCacheEntries(keysToClear, c, o, w)
 			}
 		} else {
-			// Otherwise return the cache entry corresponding to the given key.
-			keysToPrint = []string{cacheKey}
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			fmt.Fprintf(w, "Only POST is supported\n")
```

Comment:
can we return an `marshallable.Error` instead of writing raw string into the response?

Reply:
can we avoid using `fmt.Fprintf` to write into response body